api/transport/clientconfig.go

```diff
@@ -44,4 +44,5 @@ type ClientConfig interface {
 	// have already been started.
 	GetUnaryOutbound() UnaryOutbound
 	GetOnewayOutbound() OnewayOutbound
+	GetStreamOutbound() StreamOutbound
```

Comment:
We can't do this. Adding a method to an interface is a breaking change. This
was an oversight on our part when we converted ClientConfig from a struct to
an interface.

OutboundConfig was introduced to fix this, the idea being that we should use
OutboundConfig everywhere instead of ClientConfig. In case of Dispatcher, we
want to use the OutboundConfig and MustOutboundConfig methods instead of
ClientConfig, and for streaming, the client needs to attempt to upcast a
ClientConfig to OutboundConfig and error out if that's not possible.